nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Analyze/CompatabilityChecker.cs

```diff
@@ -141,15 +141,23 @@ internal static PackageReaders ReadPackage(string tempPackagePath)
                 throw new NotSupportedException($"Failed to get FindPackageByIdResource for {source.SourceUri}");
             }
 
-            var exists = await feed.DoesPackageExistAsync(
-                package.Id,
-                package.Version,
-                context.SourceCacheContext,
-                NullLogger.Instance,
-                cancellationToken);
-
-            if (!exists)
+            try
+            {
+                // misbehaving v2 apis can throw here
```

Comment:
Optional, but I do find it helpful to document more concretely where this was observed... not to throw a product under the bus, but because if someone is trying to understand exactly what we were trying to protect against or reproduce the failure they have a better pointer to it.

```suggestion
                // misbehaving v2 apis can throw here
                // We saw this happen in the wild with a user's Artifactory/Jfrog instance in July 2024.
```